R/environment.R

```diff
@@ -87,34 +87,43 @@ r_environment <- function(name, version = NULL,
   }
 
   if (is.null(custom_docker_image)) {
-    processor <- "cpu"
     if (use_gpu) {
-      processor <- "gpu"
+      base_image_with_address <- paste0("mcr.microsoft.com/azureml/base-",
+                                        "gpu:openmpi3.1.2-cuda10.0-cudnn7-",
+                                        "ubuntu16.04")
     }
-    env$docker$base_image <- paste("r-base",
-                                   processor,
-                                   sep = ":")
-    env$docker$base_image_registry$address <-
-      "viennaprivate.azurecr.io"
-  }
-
-  # if extra package is specified, generate dockerfile
-  if (!is.null(cran_packages) ||
-      !is.null(github_packages) ||
-      !is.null(custom_url_packages)) {
-    base_image_with_address <- NULL
-    registry_address <- env$docker$base_image_registry$address
-    if (!is.null(env$docker$base_image_registry$address)) {
-      base_image_with_address <- paste(registry_address,
-                                       env$docker$base_image,
-                                       sep = "/")
+    else {
+      base_image_with_address <- paste0("mcr.microsoft.com/azureml/base:",
+                                        "openmpi3.1.2-ubuntu16.04")
     }
+
     env$docker$base_dockerfile <- generate_docker_file(base_image_with_address,
                                                        cran_packages,
                                                        github_packages,
                                                        custom_url_packages)
     env$docker$base_image <- NULL
```

Comment:
If we moved this code outside if block, I am pretty sure we don't need else below.

Reply:
as discussed offline, we need this block.